packages/auth-express/src/index.ts

```diff
@@ -0,0 +1,568 @@
+  constructor(client: Client, private readonly authToken: string | undefined) {
+    this.client = this.authToken
+      ? client.withGlobals({ "ext::auth::client_token": this.authToken })
+      : client;
+  }
+
+  async isSignedIn() {
+    if (!this.authToken) return false;
+    try {
+      return await this.client.querySingle<boolean>(
+        `select exists global ext::auth::ClientTokenIdentity`
+      );
+    } catch {
+      return false;
+    }
+  }
+}
+
+export interface ExpressAuthOptions {
+  baseUrl: string;
+  authCookieName?: string;
+  pkceVerifierCookieName?: string;
+  passwordResetPath?: string;
+}
+
+type OptionalOptions = "passwordResetPath";
+
+export class ExpressAuth {
+  private readonly options: Required<
+    Omit<ExpressAuthOptions, OptionalOptions>
+  > &
+    Pick<ExpressAuthOptions, OptionalOptions>;
+  private readonly core: Promise<Auth>;
+
+  constructor(protected readonly client: Client, options: ExpressAuthOptions) {
+    this.options = {
+      baseUrl: options.baseUrl.replace(/\/$/, ""),
+      authCookieName: options.authCookieName ?? "edgedb-session",
+      pkceVerifierCookieName:
+        options.pkceVerifierCookieName ?? "edgedb-pkce-verifier",
+      passwordResetPath: options.passwordResetPath,
+    };
+    this.core = Auth.create(client);
+  }
+
+  isPasswordResetTokenValid = (resetToken: string) => {
+    return Auth.checkPasswordResetTokenValid(resetToken);
+  };
+
+  getSession = (req: ExpressRequest) => {
+    const authCookie = req.cookies[this.options.authCookieName];
+
+    return new ExpressAuthSession(this.client, authCookie);
+  };
+
+  getProvidersInfo = async () => {
+    return (await this.core).getProvidersInfo();
+  };
+
+  createSessionMiddleware = () => {
+    return async (
+      req: SessionRequest,
+      _: ExpressResponse,
+      next: NextFunction
+    ) => {
+      try {
+        req.session = this.getSession(req);
+        next();
+      } catch (err) {
+        next(err);
+      }
+    };
+  };
+
+  createBuiltinRouter = ({ callback }: { callback: RouterStack }) => {
+    const router = Router();
+
+    router.get("/signin", this.builtin.signIn);
+    router.get("/signup", this.builtin.signUp);
+    router.get("/callback", this.builtin.callback, ...callback);
+
+    return router;
+  };
+
+  createEmailPasswordRouter = (
+    routerPath: string,
+    stacks: Record<keyof typeof this.emailPassword, RouterStack>
+  ) => {
+    const router = Router();
+
+    router.post("/signin", this.emailPassword.signIn, ...stacks.signIn);
+    router.post(
+      "/signup",
+      this.emailPassword.signUp(
+        new URL(`${routerPath}/verify`, this.options.baseUrl).toString()
+      ),
+      ...stacks.signUp
+    );
+    router.get("/verify", this.emailPassword.verify, ...stacks.verify);
+    router.post(
+      "/send-password-reset-email",
+      this.emailPassword.sendPasswordResetEmail(
+        new URL(`${routerPath}/reset-password`, this.options.baseUrl).toString()
```

Comment:
Shouldn't this url be the user's reset password page, that contains a form that will then post to this `/reset-password` url?

Reply:
oop, good catch! I forgot to test the reset flow, probably some other bugs there 🙈